hack/golangci.yaml.in

```diff
@@ -180,6 +181,16 @@ linters-settings: # please keep this alphabetized
       settings:
         config: |
           {{include "hack/logcheck.conf" | indent 10 | trim}}
+  depguard:
```

Comment:
Do we need depguard for this? We already have forbidigo, which might be able to do the same.

I'm not against pulling in depguard, just wondering whether we can consolidate.

Reply:
/hold

Reply:
@pohly if we can do the same config as this does succinctly, then sure!

Reply:
we can do it as a follow up too!

Reply:
I love the approach, I don't care which impl :)

Reply:
/hold cancel

Follow-up is fine with me, I may take a look.

Reply:
One conceptual difference is that forbidigo can warn about adding new usages of cmp in files where depguard currently allows importing it.

The downside is that it has to be either a hint or all existing usages have to get a `//nolint` comment, similar to how it was done for depguard.

Probably not worth it, I just wanted to conclude the discussion.